### ISSUE TITLE
volsimpleequal bootloop after install and reboot

### ISSUE THREAD

**Issue by sashablue** [balbuze/volumio-plugins#111]
Hello,
congrats on this beautiful music server. Everything is working fine with one exception:
I installed miscellaneous / volsimpleequal which is working fine and is doing a great job.
That is until I reboot the **Raspberry PI Zero W**. I opened a ssh connection after reboot and **node** process is hogging the CPU to 95% usage and I hear the start up sound every 30 seconds or so.
Before the reboot, everything is working fine and the node & MPD processes are around using around 9-15% CPU (I use top command in ssh)
Tried several restarts to no avail.

I would attach logs if I knew which one:)

Thanks

**Comment by balbuze** [comment]
Try via ssh "sudo journalctl -f" to see where it hangs. You should see some errors when volsimpleequal starts..

**Comment by sashablue** [comment]
Hi,
i guess here is the error:
**volumio.service: main process exited, code=exited, status=1/FAILURE**
Unit volumio.service entered failed state.
Starting dynamicswap service...
Started dynamicswap service.
volumio.service holdoff time over, scheduling restart.
Starting dynamicswap service...
Started dynamicswap service.
Stopping Volumio Backend Module...
Starting Volumio Backend Module...
Started Volumio Backend Module.
**te-updater[475]: [2017-12-11 18:11:45] [info] asio async_connect error: system:111 (Connection refused)
te-updater[475]: [2017-12-11 18:11:45] [info] Error getting remote endpoint: system:107 (Transport endpoint is not connected)
te-updater[475]: [2017-12-11 18:11:45] [error] handle_connect error: Underlying Transport Error**

I attached the output of sudo journalctl -f > log.txt for more details

[log.txt](https://github.com/balbuze/volumio-plugins/files/1548903/log.txt)

**Comment by sashablue** [comment]
213

**Comment by balbuze** [comment]
should be solved with last update... Let me know as I haven't a pi0

**Comment by sashablue** [comment]
Hi balbuze,
thanks for the patch. Unfortunatelly, it does not work:(
I factory reset my last test setup for the equalizer and installed the new zip. Now it restarts OK after install, i setup the equalizer but no sound output. The counter of the song runs, but no sound. 
I logged journalctl -f starting with: playing with the plugin activated (no sound), the changed the output device back to usb  pnp sound device (and works fine but without eq), then changed the output device to loopback and again and no sound again.
The CPU usage is OK now, the web interface starts every time.
I want to specify that I from the last bug report I configured manually asound.conf, mpd.conf and the loopback interface manually and the equalizer works, but without the controlls in the webui, i just setup the eq with: alsamixer -D equal

Thank again for the hard work you are doing for this project

[log2.txt](https://github.com/balbuze/volumio-plugins/files/1565350/log2.txt)

**Comment by sashablue** [comment]
Hi again.
Something happened from my last post 15 minutes ago, trying to see if I can do about it the equalizer not working, I tried alsamixer -D equal and it gave me an error, google it and it seems that libasound2-plugin-equal was not installed. I installed it and the plugin began working. WITH VOLUME TOO:). It works after reboot too.
My sound card has hardware volume, but mixer type is set by the plugin to NONE. Yet, the volume works, i dont know if its software or hardware, but it works.

Please tell me if you want more info, but installing libasound2-plugin-equal and checking alsamixer -D equal is the single thing I've done before it started working.

Cheers

**Comment by balbuze** [comment]
Hi,
Maybe something went wrong while downloading the package. Maybe the internet connection or server down.
When you enable the plugin, the current configuration of volumio (output, mixer) is saved. The plugin then configure the output (DAC through equalizer plugin) and then restore the volume mixer. So, even if it appears as none, it is still the same hardware mixer. I'm working on this as it is a bit confusing for user. But don't try to change the output or mixer by hand or it will break the configuration of the plugin.
I'm happy if now it works for you! 
Let me know if you see improvement to be included.
Last question : what is the language you use ?  If not English or French, you can translate it. Copy /i18/strings_en.json rename it, translate and send me in order I include it!

**Comment by sashablue** [comment]
Hi,
I will test the install (on another card, this one I am keeping:)) and tell you if I had to download the dependency again.
Improvement would be a slider for software pre amp slider, because without the eq the volume lowers and even with max from the amp of the speakers, for my case, the volume is not what the speakers can output, its somewhere around 70-80%
I am romanian and momentarily I am changing jobs (I have to review the RHCSA 1100 pages). Unfortunately I dont have the time for the translation. In 2 months or so I will translate it happily to Romanian and Bulgarian too.

Thanks again for the wonderful job you are doing

PS. BUG: when I accidentally added to the queue the same item and when I removed one it removes both items.

**Comment by balbuze** [comment]
I'll add a slider to set the global volume level through the equalizer.
Waiting this, if you need you can edit index.js line 169 and increase z value (max 100)
//equalizer offset
var z = 60;
save and restart.

**Comment by sashablue** [comment]
Hi,
Made a clean install of VERSION: 2.344 and volsimpleequal and it works as it should on the RPi Zero W, I did not need to manually download the dependency libasound2-plugin-equal.
Thanks for the tip about the equalizer offset but with the new version is not needed anymore. With the eq enabled and the offset at default 60 valua the max volume is much louder. The same music piece I listened until now at 60% volume now I play it at 35% volme and it's aprox the same volume.

Cheers
